pkg/networkservice/common/mechanisms/kernel/client.go

```diff
@@ -46,12 +49,20 @@ func NewClient(opts ...Option) networkservice.NetworkServiceClient {
 }
 
 func (k *kernelMechanismClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
-	if !k.updateMechanismPreferences(request) {
+	updated, err := k.updateMechanismPreferences(request)
```

Comment:
I belive this could be simplied. 

For example,
1. rename updateMechanismPreferences -> manageMechanismPreferences
2. manageMechanismPreferences should return an error
3. move logic updated true/false into manageMechanismPreferences
```go
func (k *kernelMechanismClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption)
  if err := k.manageMechanismPreferences(request); err != nil {
  return nil, err
  }
  return next.Client(ctx).Request(ctx, req, opts)
}
// todo: simplify
func manageMechanismPreferences(request *networkservice.NetworkServiceRequest) error {
	var updated bool

	for _, m := range request.GetRequestMechanismPreferences() {
		if mechanism := kernelmech.ToMechanism(m); mechanism != nil {
			if mechanism.GetInterfaceName() == "" {
				if k.interfaceName != "" {
					mechanism.SetInterfaceName(k.interfaceName)
				} else {
					ifname, err := generateInterfaceName()
					if err != nil {
						return errors.Wrap(err, "Failed to generate kernel interface name")
					}
					mechanism.SetInterfaceName(ifname)
				}
			}
			mechanism.SetNetNSURL(netNSURL)

			updated = true
		}
	}
	if updated {
	   return
	}
  mechanism := kernelmech.ToMechanism(kernelmech.New(netNSURL))
  if k.interfaceName != "" {
	  mechanism.SetInterfaceName(k.interfaceName)
  } else {
	  ifname, err := generateInterfaceName()
	  if err != nil {
		  return errors.Wrap(err, "Failed to generate kernel interface name")
	  }
	  mechanism.SetInterfaceName(ifname)
  }
  request.MechanismPreferences = append(request.GetMechanismPreferences(), mechanism.Mechanism)
  
  return nil
}
```